src/cascadia/TerminalSettingsEditor/Appearances.h

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "Font.g.h"
+#include "Appearances.g.h"
+#include "AppearanceViewModel.g.h"
+#include "Utils.h"
+#include "ViewModelHelpers.h"
+#include "SettingContainer.h"
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    struct FontComparator
+    {
+        bool operator()(const Font& lhs, const Font& rhs) const
+        {
+            return lhs.LocalizedName() < rhs.LocalizedName();
+        }
+    };
+
+    struct Font : FontT<Font>
+    {
+    public:
+        Font(std::wstring name, std::wstring localizedName) :
+            _Name{ name },
+            _LocalizedName{ localizedName } {};
+
+        hstring ToString() { return _LocalizedName; }
+
+        WINRT_PROPERTY(hstring, Name);
+        WINRT_PROPERTY(hstring, LocalizedName);
+    };
+
+    struct AppearanceViewModel : AppearanceViewModelT<AppearanceViewModel>, ViewModelHelper<AppearanceViewModel>
+    {
+    public:
+        AppearanceViewModel(const Model::AppearanceConfig& appearance);
+
+        // background image
+        bool UseDesktopBGImage();
+        void UseDesktopBGImage(const bool useDesktop);
+        bool BackgroundImageSettingsVisible();
+
+        // font face
+        static void UpdateFontList() noexcept;
+        Windows::Foundation::Collections::IObservableVector<Editor::Font> CompleteFontList() const noexcept;
+        Windows::Foundation::Collections::IObservableVector<Editor::Font> MonospaceFontList() const noexcept;
+        bool UsingMonospaceFont() const noexcept;
+        bool ShowAllFonts() const noexcept;
+        void ShowAllFonts(const bool& value);
+
+        Windows::Foundation::Collections::IMapView<hstring, Model::ColorScheme> Schemes() { return _Schemes; }
+        void Schemes(const Windows::Foundation::Collections::IMapView<hstring, Model::ColorScheme>& val) { _Schemes = val; }
+
+        WINRT_PROPERTY(IHostedInWindow, WindowRoot, nullptr);
+
+        // These settings are not defined in AppearanceConfig, so we grab them
+        // from the source profile itself. The reason we still want them in the
+        // AppearanceViewModel is so we can continue to have the 'Text' grouping
+        // we currently have in xaml, since that grouping has some settings that
+        // are defined in AppearanceConfig and some that are not.
+        OBSERVABLE_PROJECTED_SETTING(_appearance.SourceProfile(), FontFace);
+        OBSERVABLE_PROJECTED_SETTING(_appearance.SourceProfile(), FontSize);
+        OBSERVABLE_PROJECTED_SETTING(_appearance.SourceProfile(), FontWeight);
```

Comment:
When we add unfocused appearance to the SUI, we will set a bool on these settings in xaml to only be shown for the default appearance (we do not want to add these settings to AppearanceConfig because we had concerns about causing a resize when transitioning between focused/unfocused states)